### PR TITLE
New integer exponent pow 

### LINKED ISSUE #8704: Fast runtime integer power
### Description of the enhancement or error report
Add a fast `MathUtils::pow` function that allows runtime selectable integer powers.

### Rationale for the enhancement or information for reproducing the error
Some model free energies with tunable spinodal region contain high integer power terms (80th power). This can be made fast by looking at the bits of the integer exponent and multiplying progressively squared values for the set bits.

### Identified impact
Added capability.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #8704

### CODE REVIEW

**Comment by friedmud** [comment]
Have you tested to see whether your `pow()` is faster than the template metaprogramming `pow()` in libMesh: https://github.com/libMesh/libmesh/blob/master/include/utils/utility.h#L194

**Comment by dschwen** [comment]
Ha, of course not. It will be slower, but the template version needs the exponent specified at compile time and I need a version that takes the exponent at run time!

**Comment by friedmud** [comment]
Gotcha - makes sense!

File: modules/phase_field/src/utils/MathUtils.C [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "MathUtils.h"
+#include "libmesh/utility.h"
+
+namespace MathUtils
+{
+
+Real poly1Log(Real x, Real tol, int deriv)
```
Comment (andrsd) [inline modules/phase_field/src/utils/MathUtils.C]: return type on a new line, and below as well...

File: modules/phase_field/src/utils/MathUtils.C [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+{
+  Real c1 = 1.0/tol;
```
Comment (andrsd) [inline modules/phase_field/src/utils/MathUtils.C]: Spaces around `/` and below as well...

File: modules/phase_field/src/utils/MathUtils.C [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  Real c2 = std::log(tol) - 1.0;
+
+  Real value=0.0;
```
Comment (andrsd) [inline modules/phase_field/src/utils/MathUtils.C]: Spaces around `=` and below in the other function(s).

File: modules/phase_field/src/utils/MathUtils.C [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+  if (deriv == 0)
+  {
+    if (x < tol)
+      value = c1*x + c2;
```
Comment (andrsd) [inline modules/phase_field/src/utils/MathUtils.C]: Spaces around `*` and below as well...

File: modules/phase_field/src/utils/MathUtils.C [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+{
+  Real c1=-0.5/(tol*tol);
+  Real c2=2.0/tol;
+  Real c3=std::log(tol) - 3.0/2.0;
+
+  Real value=0.0;
+
+  if (deriv == 0)
+  {
+    if (x<tol)
```
Comment (andrsd) [inline modules/phase_field/src/utils/MathUtils.C]: Spaces around `<`.

File: unit/include/MathUtilsTest.h [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+#ifndef MATHUTILSTEST_H
+#define MATHUTILSTEST_H
+
+//CPPUnit includes
+#include "GuardedHelperMacros.h"
+
+class MathUtilsTest : public CppUnit::TestFixture
+{
+
+  CPPUNIT_TEST_SUITE( MathUtilsTest );
```
Comment (andrsd) [inline unit/include/MathUtilsTest.h]: This is a copy-n-paste, but it is against the code standard. The code formatting script would remove those space...

The same thing in `.C`.

**Review by andrsd — CHANGES_REQUESTED** [review]
There are lots of operators missing spaces around them. I understand that it is a copy-n-paste thing and it does make the code a little bit more legible (somewhere), but the anticipated code formating script will undo that.

**Comment by dschwen** [comment]
I'll run clang-format over this, but those are all old issue unrelated to this PR.

**Review by andrsd — APPROVED** [review]

**Comment by permcody** [comment]
One overall comment: @rwcarlsen suggested that we move towards lower case namespace names. His argument is that it's more standard, i.e all the standard namespaces are lowercase and it helps to differentiate a namespace from a class name. I think it's actually a pretty good idea. This utilities is in MathUtils which is fine for the time being. Just starting the discussion.

**Comment by dschwen** [comment]
Yeah, we might want to have that discussion in a separate ticket, but I am in favor of UpperCase namespace names. It differentiates MOOSE code from standard lib code, it makes them look like static class members. But I don't feel strongly about it.

**Pull request merged**